mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLHostOps.td

```diff
@@ -123,6 +123,24 @@ def SYCLHostHandlerSetNDRange : SYCL_HostOp<"handler.set_nd_range"> {
   let hasVerifier = 1;
 }
 
+def SYCLHostSetCaptured : SYCL_HostOp<"set_captured"> {
+  let summary = "Marks a value as captured by a kernel function object.";
+  let description = [{
+    This operation represents that the given `value` was captured in the kernel
+    function object `lambda` at index `index`. The op is created during the 
+    progressive raising towards the `sycl.host.launch_kernel` op.
+
+    Example:
+    ```
+    sycl.host.set_captured %lambda[2] = %accessor : !llvm.ptr, !llvm.ptr
+    ```
+  }];
+  let arguments = (ins LLVM_AnyPointer:$lambda, I64Attr:$index, AnyType:$value);
+  let assemblyFormat = [{
+    $lambda `[` $index `]` `=` $value attr-dict `:` type(operands)
```

Comment:
Nit: Do we need the type of the `lambda` argument in the assembly format? I would expect it to always be `!llvm.ptr` with opaque pointers.

Reply:
Right, no, will remove it.

Reply:
Question: do we need to know whether the `value` was captured by value or reference ?

Reply:
Per SYCL specification, only capture by value is possible for kernel function objects.

Reply:
Done. I didn't see a buildable version or alias of the `!llvm.ptr` type upstream, so I introduced this here for now.

Reply:
Discussed with @sommerlukas that leaving out the type is not worth the haphazard "extension" of the LLVM dialect for our dialect.